app/src/test/java/com/davidtomas/taskyapp/features/auth/presentation/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,165 @@
+package com.davidtomas.taskyapp.features.auth.presentation.login
+
+import app.cash.turbine.test
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.davidtomas.taskyapp.MainCoroutineExtension
+import com.davidtomas.taskyapp.R
+import com.davidtomas.taskyapp.core.domain._util.EMPTY_STRING
+import com.davidtomas.taskyapp.core.domain._util.Result
+import com.davidtomas.taskyapp.core.domain.model.DataError
+import com.davidtomas.taskyapp.core.presentation.util.UiText
+import com.davidtomas.taskyapp.features.auth.domain.AuthRepository
+import com.davidtomas.taskyapp.features.auth.domain.model.AuthModelStubs
+import com.davidtomas.taskyapp.features.auth.domain.useCase.LoginUseCase
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(MainCoroutineExtension::class)
+class LoginViewModelTest {
+
+    private lateinit var loginViewModel: LoginViewModel
+    private lateinit var loginUseCase: LoginUseCase
+    private lateinit var authRepository: AuthRepository
+
+    @BeforeEach
+    fun setUp() {
+        authRepository = mockk()
+        loginUseCase = LoginUseCase(
+            authRepository = authRepository
+        )
+        loginViewModel = LoginViewModel(
+            loginUseCase = loginUseCase
+        )
+    }
+
+    @Test
+    fun `when onAction receives OnChangePasswordVisibility action password visibility is changed`() {
+        val initialPasswordVisibility = loginViewModel.state.isPasswordVisible
+        loginViewModel.onAction(LoginAction.OnChangePasswordVisibility)
+        assertTrue(loginViewModel.state.isPasswordVisible != initialPasswordVisibility)
```

Comment:
Could be rephrased to `assertThat(loginViewModel.state.isPasswordVisible).isNotEqualTo(initialPasswordVisibility)`